### PR TITLE
Fix Static Templates Not Found

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Fix the addsensor html page not found by fixing the path of the templates directory for sensorsafrica. Additionally, introduced whitenoise to compress and serve static files.

Moreover, this fixes the path the the templates folder which seems to not work without specifying it in TEMPLATES DIRS.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] Merge #11 
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: sensorsafrica/settings.py [hunk 1]
```diff
@@ -70,15 +70,16 @@
 TEMPLATES = [
     {
         "BACKEND": "django.template.backends.django.DjangoTemplates",
-        "DIRS": [os.path.normpath(os.path.join(SITE_ROOT, "templates"))],
+        "DIRS": [os.path.normpath(os.path.join(BASE_DIR, "sensorsafrica/templates"))],
```
Comment (kilemensi) [inline sensorsafrica/settings.py]: Do we need to specify `default` template folders too?

File: sensorsafrica/settings.py [hunk 2]
```diff
@@ -136,3 +114,16 @@
...
 
 STATIC_URL = "/static/"
+
+STATIC_ROOT = "static"
+
+# Extra places for collectstatic to find static files.
+STATICFILES_FINDERS = [
+    "django.contrib.staticfiles.finders.FileSystemFinder",
+    "django.contrib.staticfiles.finders.AppDirectoriesFinder",
+]
```
Comment (kilemensi) [inline sensorsafrica/settings.py]: Motivate why we need these.

Comment (karimkawambwa) [reply sensorsafrica/settings.py]: The idea is that moving forward any app with static files will have to create a static folder in the add with the html code for example.

Comment (kilemensi) [reply sensorsafrica/settings.py]: @karimkawambwa, these appear to be the default in Django: https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-STATICFILES_FINDERS. Why do we still need to include them here?

Comment (karimkawambwa) [reply sensorsafrica/settings.py]: Thats valid reason, i missed that they are default,

File: sensorsafrica/settings.py (hunk 1 above)
Comment (kilemensi) [reply sensorsafrica/settings.py]: All other apps that we're using, we don't define `DIRS` at all and it works. `templates` is a known location.

I think the problem is here: https://github.com/CodeForAfricaLabs/sensors.AFRICA-api/blob/912defb7d8fa8a1869499089bf735f33c579a96b/sensorsafrica/settings.py#L48-L50

You haven't included `"sensorsafrica"` app and hence `"sensorsafrica/templates"` is not searched for templates.

**Comment by karimkawambwa** [comment]
@kilemensi 
> Explain it a bit... Looking at default [WhiteNoise](http://whitenoise.evans.io/en/stable/django.html) docs, you seem to be adding a bit more that the minimum required to make it work.

This fixes the path the the templates folder which seems to not work without specifying it in TEMPLATES DIRS.

File: sensorsafrica/settings.py (hunk 1 above)
Comment (karimkawambwa) [reply sensorsafrica/settings.py]: that does resolve it

**Pull request merged**